File: src/BreakfastProvider.Api/Models/Requests/WaffleRequest.cs
namespace BreakfastProvider.Api.Models.Requests;

public record WaffleRequest
{
    public string? Milk { get; init; }

    public string? Flour { get; init; }

    public string? Eggs { get; init; }

    public string? Butter { get; init; }

    public List<string> Toppings { get; init; } = [];
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/model/request/WaffleRequest.java
package io.lemonlion.breakfast.model.request;

import java.util.ArrayList;
import java.util.List;

/** Twin of C# {@code WaffleRequest}. */
public record WaffleRequest(String milk, String flour, String eggs, String butter, List<String> toppings) {

    public WaffleRequest {
        if (toppings == null) {
            toppings = new ArrayList<>();
        }
    }
}
